enhancements/microshift/microshift-router-configuration.md

```diff
@@ -0,0 +1,456 @@
+   everything according to it.
+3. The router will enabled/disabled, be exposed on the specified ports, on the
+   specified IPs, allowing only specified IP addresses, all according to the
+   cluster admin provided configuration. This includes the status in the
+   LoadBalancer service type for the router.
+
+### API Extensions
+As described in the proposal, there is an entire new section in the configuration:
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+  expose:
+    - <IP address/hostname> # Defaults to the node's exposed IP addresses.
+```
+
+For more information check each individual section.
+
+### Implementation Details
+The default router is composed of a bunch of assets part of the MicroShift
+binary. These assets come from the rebase, copied from the original router in
+[cluster-ingress-operator](https://github.com/openshift/cluster-ingress-operator).
+There is already a [LoadBalancer service](https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/manifests/assets/router/service-cloud.yaml)
+included, which MicroShift does not copy yet. Depending on the options
+configured in MicroShift, this resource will need further customization done
+by MicroShift's start process.
+
+The rest of the changes imply only additions. The LoadBalancer controller needs
+an expansion on its capabilities to include more IPs in the status and log any
+changes to services. All the logic to control NetworkPolicy resources is also
+new.
+
+#### How config options change manifests
+Each of the configuration options described above has a direct effect on the
+manifests that MicroShift will apply after starting.
+
+The `router.status` option will drive whether the router `Deployment` and the
+`Service` get created.
+```yaml
+ingress:
+  status: <Enabled|Disabled> # Defaults to Enabled.
+```
+
+The `router.policy.ports.http` and `router.policy.ports.https` will determine,
+within the `Deployment` and the `Service`, which ports get configured to be
+exposed.
+```yaml
+ingress:
+  ports:
+    http: <int> # Defaults to 80.
+    https: <int> # Defaults to 443.
+```
+
+The `router.policy.expose` option, which is a list, will configure the
+`status.loadBalancer` field in the `Service`. Ovnk will pick up this field to
+configure the iptables rules.
+```yaml
+ingress:
+  expose:
+    - <IP address/hostname> # Defaults to the node's exposed IP addresses.
+```
+
+### Risks and Mitigations
+Some of the features in this enhancement proposal rely on the
+[LoadBalancer controller](loadbalancer-service-support.md).
+Some of the features rely on NetworkPolicy resources, so the CNI needs
+to support that.
+
+### Drawbacks
+Some of the features depend on a non-agnostic CNI design, which is the
+`LoadBalancer` controller. A different behavior should be expected if using a
+different CNI.
+
+## Design Details
+#### Why LoadBalancer service
```

Comment:
I think lines 280-319 are saying

We used host ports on the router originally because that allowed clients from anywhere to communicate with the router before OVN-K had a way to expose something as a load balancer service. Using host ports meant CRI set up access to the ports, using iptables rules that bypass firewalld and cannot be controlled with the NetworkPolicy API, so it is not currently possible to control access to the router at all. As part of this enhancement, we will change the service type from host port to load balancer. That will let us easily configure the ports used by the router by setting the ports on the service, and let a user apply NetworkPolicy to control access to the router.

Is that right?

Reply:
Yes.
Back when host ports were introduced there was no support for load balancer services and host network was not good enough, so this was the only way out. Even if host network would have sufficed, it is subject to the same limitations than host ports.
The rest of what you said is 100% correct.

Reply:
Maybe you can tighten up this section, since it's mostly history.

Reply:
Done